example/commitpr/main.go

```diff
@@ -0,0 +1,205 @@
+	"golang.org/x/oauth2"
+)
+
+var (
+	sourceOwner   = flag.String("source-owner", "", "Name of the owner (user or org) of the repo to create the commit in.")
+	sourceRepo    = flag.String("source-repo", "", "Name of repo to create the commit in.")
+	commitMessage = flag.String("commit-message", "", "Content of the commit message.")
+	commitBranch  = flag.String("commit-branch", "", "Name of branch to create the commit in. If it does not already exists, it will be created using the `base-branch` parameter")
+	baseBranch    = flag.String("base-branch", "master", "Name of branch to create the `commit-branch` from.")
+	prRepoOwner   = flag.String("merge-repo-owner", "", "Name of the owner (user or org) of the repo to create the PR against. If not specified, the value of the `-source-owner` flag will be used.")
+	prRepo        = flag.String("merge-repo", "", "Name of repo to create the PR against. If not specified, the value of the `-source-repo` flag will be used.")
+	prBranch      = flag.String("merge-branch", "master", "Name of branch to create the PR against (the one you want to merge your branch in via the PR).")
+	prSubject     = flag.String("pr-title", "", "Title of the pull request. If not specified, no pull request will be created.")
+	prDescription = flag.String("pr-text", "", "Text to put in the description of the pull request.")
+	sourceFiles   = flag.String("files", "", "Comma-separated list of files to commit and their location. The local file is separated by his target location by a semi-colon. If the file should be in the same location with the same name, you can just put the file name and omit the repetition. Example: README.md,main.go:github/examples/commitpr/main.go`")
+	authorName    = flag.String("author-name", "", "Name of the author of the commit.")
+	authorEmail   = flag.String("author-email", "", "Email of the author of the commit.")
+)
+
+var client *github.Client
+var ctx = context.Background()
+
+// getRef returns the commit branch reference object if it exists or creates it from the base branch before returning it
+func getRef() (ref *github.Reference, err error) {
+	if ref, _, err = client.Git.GetRef(ctx, *sourceOwner, *sourceRepo, "refs/heads/"+*commitBranch); err == nil {
+		return
+	}
+	// We consider that an error means the branch has not been found and needs to be created
+	if *commitBranch != *baseBranch {
+		if *baseBranch == "" {
+			log.Fatal("The `-base-branch` should not be set to an empty string when the branch specified by `-commit-branch` does not exists")
+		}
+		var baseRef *github.Reference
+		if baseRef, _, err = client.Git.GetRef(ctx, *sourceOwner, *sourceRepo, "refs/heads/"+*baseBranch); err != nil {
+			return
+		}
+		newRef := &github.Reference{Ref: github.String("refs/heads/" + *commitBranch), Object: &github.GitObject{SHA: baseRef.Object.SHA}}
+		if ref, _, err = client.Git.CreateRef(ctx, *sourceOwner, *sourceRepo, newRef); err != nil {
+			return
+		}
+	}
+	return
```

Comment:
This is slightly odd here because it could potentially return `nil, nil` or `ref, nil` depending on the args and results of the `if` statements. If it were me, I would like to isolate the `nil, nil` case and instead turn that into an error, and then change this line to explicitly be `return ref, nil`.

Maybe if you process the args in `main`, you wouldn't have to do that in this function.